infra/config.go

```diff
@@ -34,3 +34,63 @@ type StateNode struct {
 	// KeyPath defines the location of the SSH key
 	KeyPath string `json:"key_path"`
 }
+
+// AWSConfig describes AWS EC2 test configuration
+type AWSConfig struct {
+	AccessKey string `json:"access_key" yaml:"access_key" validate:"required"`
+	SecretKey string `json:"secret_key" yaml:"secret_key" validate:"required"`
+	// Region specifies the EC2 region to install into
+	Region string `json:"region" yaml:"region" validate:"required"`
+	// KeyPair specifies the name of the SSH key pair to use for provisioning
+	// nodes
+	KeyPair string `json:"key_pair" yaml:"key_pair" validate:"required"`
+	// VPC defines the Amazon VPC to install into.
+	// Specify "Create new" to create a new VPC for this test run
+	VPC string `json:"vpc" yaml:"vpc" validate:"required"`
+	// KeyPath specifies the location of the SSH key to use for remote access.
+	// Mandatory only with terraform provisioner
+	SSHKeyPath string `json:"key_path" yaml:"key_path"`
+	// SSHUser defines SSH user used to connect to the provisioned machines
+	SSHUser string `json:"ssh_user" yaml:"ssh_user" validate:"required"`
+	// InstanceType defines the type of AWS EC2 instance to boot.
+	// Relevant only with terraform provisioner.
+	// Defaults are specific to the terraform script used (if any)
+	InstanceType string `json:"omitempty,instance_type" yaml:"instance_type"`
+	// ExpandProfile specifies an optional name of the server profile for AWS expand operation.
+	// If the profile is unspecified, the test will use the first available.
+	ExpandProfile string `json:"expand_profile" yaml:"expand_profile"`
+	// ExpandAwsInstanceType specifies an optional instance type for AWS expand operation
+	ExpandAWSInstanceType string `json:"expand_instance_type" yaml:"expand_instance_type"`
+	// ClusterName defines tagging and placement group for resources allocated
+	ClusterName string `json:"cluster_name" yaml:"cluster_name" validate:"required"`
+}
+
+// FIXME : replace with embedded validation rules
+func (r *AWSConfig) IsEmpty() bool {
+	return r.AccessKey == "" && r.SecretKey == ""
+}
+
+// AzureConfig specifies Azure cloud specific parameters
+// see https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal
+// to obtain subscription_id, client_id, client_secret and tenant_id
+type AzureConfig struct {
+	SubscriptionId string `json:"subscription_id" yaml:"subscription_id" validate:"required"`
```

Comment:
Still, it makes for nicer autogenerated docs to follow the idiom of 'FooBar ...' comments on every exported `FooBar`